components-api/src/main/scala/pl/touk/nussknacker/engine/api/definition/ParameterValidator.scala

```diff
@@ -38,12 +42,48 @@ trait Validator {
 
 //TODO: These validators should be moved to separated module
 
+case class SpelExpressionParameterValidator(
+    validationExpression: String,
+    validationFailedMessage: Option[String]
+) extends ParameterValidator {
+
+  private val parser: ExpressionParser   = new SpelExpressionParser()
+  private val parsedValidationExpression = parser.parseExpression(validationExpression)
+
+  override def isValid(paramName: String, value: Any, label: Option[String])(
+      implicit nodeId: NodeId
+  ): Validated[PartSubGraphCompilationError, Unit] = {
+    val context: StandardEvaluationContext = new StandardEvaluationContext()
+    context.setVariable(paramName, value)
```

Comment:
What about the situation when `paramName` is not the correct variable name? Wouldn't be better to use just the `#value`?

Reply:
`#value` would be fine but using `paramName` allows us to validate parameters against each other. For example, enricher has two params: `hourFrom` and `hourTo`. You can write validation expression for the first one:
`#hourFrom >= 0 && #hourFrom < 24` 
and for second one:
`#hourTo >= 0 && #hourTo < 24 && #hourFrom < #hourTo`

Reply:
It is an interesting idea and we can go in this direction, but it requires to do some variable name sanitization

Reply:
Name sanitization should be done when defining fragment parameters, not when we are trying to fill them and validate, WDYT?

Reply:
I think that name validation should be probably done when defining fragment parameters. Here we are loosely coupled with fragments so we need to handle situations when these names are invalid variable names. Let's consider validation of some other parameters like `Topic name`

Reply:
You're right. I've replaced that with `#value` and added "TODO".

Reply:
Done :)